Desktop/data/importers/excel/excel.cpp

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright (C) 2013-2024 University of Amsterdam
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+#include "excel.h"
+#include "utilities/qutils.h"
+#include "utils.h"
+
+#include <QFileInfo>
+#include <QDebug>
+
+using namespace std;
+
+Excel::Excel(const string &locator)
+{
+	_path = locator;
+}
+
+void Excel::open()
+{
+	_fileSize = Utils::getFileSize(_path);
+
+	if (_fileSize < 0)
+		throw runtime_error("Could not access file");
+
+	if (_fileSize == 0)
+		throw runtime_error("File is empty");
+}
+void Excel::openWorkbook() 
+{
+	QString xlsFilePath = QString::fromStdString(_path);
+	const char* utf8Path = xlsFilePath.toUtf8();
+	QString extension = QFileInfo(xlsFilePath).suffix().toLower();
```

Comment:
```suggestion
	QString xlsFilePath = tq(_path);
	const char* utf8Path = _path.c_str(); //But it would be better to just use _path.c_str() directly if you need it. It is in utf8 in any case.
	QString extension = QFileInfo(xlsFilePath).suffix().toLower();
```